[2단계 미션] 박예은 미션 제출합니다

ye6194 opened the pull request:
## CategoryFilter
![image](https://github.com/user-attachments/assets/778ce2b0-65eb-4a44-af5a-c053a964b4bf)
CategoryFilter에서 카테고리를 바꾸면 handleChange()가 실행되고, handleChange()가 onChangeCategory()를 실행시켜 `category`를 바꿉니다.

<br/>

## App
![image](https://github.com/user-attachments/assets/aae2dea0-e3bc-489e-b78b-b814f43819a7)
useEffect()를 이용해 category가 바뀌면 전체 음식점 중에서 사용자가 고른 카테고리의 음식점만 배열에 들어가게 했습니다.

<br/>

## RestaurantList
![image](https://github.com/user-attachments/assets/1db47f1c-1cac-4139-995e-f85ba1bc7fbc)
map()을 이용해 필터된 음식점 목록을 나타냅니다. categoryInEnglish 이걸 생각하느라 애먹었어요...

<br/>

## Components Structure
![image](https://github.com/user-attachments/assets/c390d593-49b1-434d-a1d1-d6030c50a74f)

<br/>

## 궁금한점
- 저는 useEffect를 썼는데 왠지 useEffect를 쓰지 않아도 될 것 같긴 합니다. 좋은 방법이 있을까요?? 리뷰어님은 어떻게 하셨나요?
- CategoryFilter.jsx에서 `console.log(category);` 이 부분이 4번이나 실행되는 이유가 뭘까요??
![image](https://github.com/user-attachments/assets/602784c6-aeed-4f1d-9e9f-767967e12167)

## 느낀점
확실히 1단계보다 어려웠어요. 생각보다 난이도가 있는느낌...? 제가 너무 어렵게 생각해서 그런 거일 수도 있어요!

<br/>
<br/>

(Merge branch 'step1'~까지가 step1 구현, 그 이후부터가 step2 구현입니다!!)

ye6194 committed:
feat: 초기세팅

ye6194 committed:
feat: 초기세팅

ye6194 committed:
feat: 컴포넌트 그리기

ye6194 committed:
feat: 스타일 적용

ye6194 committed:
Merge branch 'step1' of https://github.com/ye6194/self-paced-react into step1

ye6194 committed:
feat: 카테고리 필터에 따라 필터된 음식점 목록을 보여줌

ye6194 committed:
feat: 필터된 음식점 목록을 보여줄 때 카테고리에 맞는 이미지 출력

ye6194 committed:
chore: 불필요한 주석 제거

Indigochi1d commented on src/App.jsx:
```diff
@@ -1,7 +1,76 @@
+      description: "멕시칸 캐주얼 그릴",
+      category: "기타",
+    },
+  ];
```
step1에서 리뷰했는데 이미 step2에선 이미 적용이 되어있었네요😅

Indigochi1d commented on src/App.jsx:
```diff
@@ -1,7 +1,76 @@
 function App() {
-  return <h1>Self-Paced React</h1>;
+  const [category, setCategory] = useState("전체");
+  const [filteredRestaurants, SetFilteredRestaurants] = useState([]);
```
setter함수의 첫글자가 소문자이어야할 것 같아요

ye6194 replied:
헉 오타났네요🫢

Indigochi1d commented on src/App.jsx:
```diff
@@ -1,7 +1,76 @@
+    },
+  ];
+
+  useEffect(() => {
+    if (category === "전체") SetFilteredRestaurants(restaurants);
+    else
+      SetFilteredRestaurants(restaurants.filter((restaurant) => restaurant.category === category));
+  }, [category]);
```
이런 방식으로 하셨군요!!  저는 생각도 못했는데 이렇게도 할 수 있네요.
제 코드를 리뷰하실때 보시겠지만 아래의 흐름도를 따라가보시면 보이실거에요!

간략하게 설명드리자면 저는 setter함수를 던져주는 방식으로 사용했습니다
![image](https://github.com/user-attachments/assets/eeefdfa6-600f-4b1b-a733-9259474baba2)

ye6194 replied:
이런 방법도 있군요!!

Indigochi1d commented on src/components/AddRestaurantModal.jsx:
```diff
@@ -0,0 +1,49 @@
+            {/* 카테고리 */}
+            <div className="form-item form-item--required">
+              <label htmlFor="category text-caption">카테고리</label>
+              <select name="category" id="category" required>
+                <option value="">선택해 주세요</option>
+                <option value="한식">한식</option>
+                <option value="중식">중식</option>
+                <option value="일식">일식</option>
+                <option value="양식">양식</option>
+                <option value="아시안">아시안</option>
+                <option value="기타">기타</option>
+              </select>
```
이 부분도 map함수로 표현되면 더 좋을 것 같아요!

ye6194 replied:
한번 수정해보겠습니다!

Indigochi1d commented on src/components/CategoryFilter.jsx:
```diff
@@ -0,0 +1,33 @@
+          value={category}
+          onChange={handleChange}
+        >
+          <option value="전체">전체</option>
+          <option value="한식">한식</option>
+          <option value="중식">중식</option>
+          <option value="일식">일식</option>
+          <option value="양식">양식</option>
+          <option value="아시안">아시안</option>
+          <option value="기타">기타</option>
```
마찬가지로 map함수의 적용이 좋아보입니다!

Indigochi1d commented on src/components/CategoryFilter.jsx:
```diff
@@ -0,0 +1,33 @@
+  function handleChange(event) {
+    onChangeCategory(event.target.value);
+  }
+  console.log(category);
```
console.log는 지워주시는게 좋을 것 같아요!

ye6194 replied:
넵넵

Indigochi1d commented on src/components/RestaurantList.jsx:
```diff
@@ -0,0 +1,38 @@
+    <>
+      <section className="restaurant-list-container">
+        <ul className="restaurant-list">
+          {restaurants.map((restaurant) => (
+            <li key={restaurant.id} className="restaurant">
+              <div className="restaurant__category">
+                <img
+                  src={`/category-${categoryInEnglish[restaurant.category]}.png`}
+                  alt="한식"
+                  className="category-icon"
+                />
+              </div>
+              <div className="restaurant__info">
+                <h3 className="restaurant__name text-subtitle">{restaurant.name}</h3>
+                <p className="restaurant__description text-body">{restaurant.description}</p>
```
map함수의 적용과 jsx안에서의 표현식까지 좋습니다!!

Indigochi1d reviewed (commented):
step2의 리뷰를 끝냈습니다!! 이번 미션에서는 카테고리 선택에 따른 정렬을 요구했는데요. 이 과정에서 useState를 사용한 상태의 선언과 props로 setter함수를 넘기며 컴포넌트가 데이터를 받아들이는 방식까지 사소하지만 중요한 것들을 배울 수 있었던 미션이었던 것 같아요!

특히 useEffect에 의존성 배열을 category로 추가하면서 렌더링하는 방식은 생각해보지 못했는데 한 수 배워갑니다..ㅎ

이번주 미션도 고생하셨고 중간고사 끝나고 뵈어요! 궁금한 점이 있으시거나 의논해보고 싶은게 있으시다면 꼭 멘션주세요!!

ye6194 committed:
refactor: data를 다른 파일로 분리

- import 해서 map() 사용

wzrabbit commented on src/data/categoryData.js:
```diff
@@ -0,0 +1,3 @@
+const categoryData = ["한식", "중식", "일식", "양식", "아시안", "기타"];
+
+export default categoryData;
```
`categoryData`는 역할 특성상 런타임 동안 바뀔 일이 없고, 바뀌어서는 안 되고, 그리고 `export` 되어 여러 곳에서까지 쓰일 수 있는 변수네요?

이 변수에 대한 정보를 모르는 개발자의 입장에서는, 이 변수를 import한 다음, 원치 않는 방법으로 변경할 수도 있어보이고, 변경되어서는 안 되는 값임을 알려주기 위한 수단도 부족해 보인다고 생각하는데, 해결할 수 있는 방법은 없을까요?

ye6194 replied:
음... 읽자마자 readonly가 떠올랐는데 타입스크립트에만 있고 자바스크립트에는 없어서 아쉽네요🫠

찾아보니 Object.freeze()라는게 있어서 이렇게 써봤는데 의천님이 보시기에는 어떤가요??
```javascript
export const categoryData = Object.freeze([
  "전체",
  "한식",
  "중식",
  "일식",
  "양식",
  "아시안",
  "기타",
]);
```

wzrabbit replied:
좋은 방법인 것 같아요 👍🏻 저도 사실 `Object.freeze()`를 생각했어요 😄 철통 보안이네요
그리고 이 이야기도 했었는데

> 변경되어서는 안 되는 값임을 알려주기 위한 수단도 부족해 보인다고 생각하는데

이 경우에는 제가 떠올린 방법은 이 변수가 **상수**라는 것을 알려주는 것이라 생각해요. `UPPER_SNAKE_CASE`

ye6194 replied:
`Object.freeze()`가 변경되어서는 안 되는 값임을 알려주는 역할도 한다고 생각했는데 `UPPER_SNAKE_CASE`를 써서 선언하는 방법도 있었네요😮

wzrabbit commented on src/components/CategoryFilter.jsx:
```diff
@@ -0,0 +1,36 @@
+          {categoryData.map((category) => (
+            <option key={category}>{category}</option>
+          ))}
+          {/* <option value="전체">전체</option>
+          <option value="한식">한식</option>
+          <option value="중식">중식</option>
+          <option value="일식">일식</option>
+          <option value="양식">양식</option>
+          <option value="아시안">아시안</option>
+          <option value="기타">기타</option> */}
```
```suggestion
```

더 이상 안 쓰이는 코드라면, 삭제를 부탁드려도 괜찮을까요 😊♻️

ye6194 replied:
아이고 이걸 확인 못했네요..! 지우겠습니다

wzrabbit commented on src/App.jsx:
```diff
@@ -1,7 +1,37 @@
+    if (category === "전체") SetFilteredRestaurants(restaurantsData);
+    else
+      SetFilteredRestaurants(
+        restaurantsData.filter((restaurant) => restaurant.category === category)
```
`Array.prototype.filter` 👍🏻 

특정 조건을 만족하는 값들만 걸러내기 위한 목적으로도 적합하고, 동작도 예측하기 쉽고, 읽기도 좋은 좋은 고차함수의 사용 예시네요~

wzrabbit commented on src/components/AddRestaurantModal.jsx:
```diff
@@ -0,0 +1,47 @@
+    <>
+      <div className="modal modal--open">
+        <div className="modal-backdrop"></div>
+        <div className="modal-container">
+          <h2 className="modal-title text-title">새로운 음식점</h2>
+          <form>
+            {/* 카테고리 */}
+            <div className="form-item form-item--required">
+              <label htmlFor="category text-caption">카테고리</label>
+              <select name="category" id="category" required>
+                <option value="">선택해 주세요</option>
+                {categoryData.map((category) => (
+                  <option key={category}>{category}</option>
+                ))}
+              </select>
+            </div>
+
+            {/* 음식점 이름 */}
+            <div className="form-item form-item--required">
+              <label htmlFor="name text-caption">이름</label>
+              <input type="text" name="name" id="name" required />
+            </div>
+
+            {/* 설명 */}
+            <div className="form-item">
+              <label htmlFor="description text-caption">설명</label>
+              <textarea name="description" id="description" cols="30" rows="5"></textarea>
+              <span className="help-text text-caption">메뉴 등 추가 정보를 입력해 주세요.</span>
+            </div>
+
+            {/* 추가 버튼 */}
+            <div className="button-container">
+              <button className="button button--primary text-caption">추가하기</button>
+            </div>
+          </form>
+        </div>
+      </div>
+    </>
+  );
+}
```
이 컴포넌트는 주석으로도 세부 역할을 나누신 것을 보시면 아시겠지만, 하나의 컴포넌트가 여러 종류의 메뉴나 역할을 담고 있는 것 같아 보이네요?

이 경우라면 이제 필요에 따라 내부의 메뉴들을 더 작은 단위의 컴포넌트로도 나눌 수 있어 보입니다, 1단계에서는 그게 힘들었지만, 이제 state와 props를 사용할 수 있으니, 공통점이 많은 UI가 있거나 더 작은 컴포넌트로 분할해 복잡성을 분배해 줄 수 있는 경우가 있다면 충분히 나눠볼 수 있을 거에요

이번 미션에서는 반영하지 않으셔도 좋고요, 앞으로도 더 많은 단계의 미션이 더 남아 있으니, 예은님의 컴포넌트를 생각하는 가치관에 따라 여러 시도를 해 보시기를 바랍니다

결론이 "굳이 나눠야 하나, 안 나눌래!" 여도 좋아요. 아마 나눌 지, 나누지 않을 지 저울질을 하고 계실 수도 있을 거에요. 어느 쪽이든 좋지만, 충분히 고민을 하시고, 이유를 남겨두신 채로 구현해 보시면 정말 좋은 연습이자 경험이 될 수 있을 거에요

ye6194 replied:
저는 여기서 하나도 나누지 않았지만, 작게 나눈다면 카테고리, 음식점 이름, 설명, 추가 버튼으로 나눌 수 있겠네요. 저는 만약에 나누게 된다면 카테고리, 음식점 이름, 설명 / 추가 버튼 이렇게 나눌 것 같아요. 좋은 조언 감사합니다!!😊

wzrabbit commented on src/components/RestaurantList.jsx:
```diff
@@ -0,0 +1,38 @@
+import "../styles/RestaurantListStyle.css";
+
+function RestaurantList({ restaurants }) {
+  const categoryInEnglish = {
+    한식: "korean",
+    중식: "chinese",
+    일식: "japanese",
+    양식: "western",
+    아시안: "asian",
+    기타: "etc",
+  };
```
이 변수는 컴포넌트 밖에 선언되면 더 좋을 것 같은데, 어떻게 생각하시나요? 해당 변수는 어떤 state나 props의 영향을 받지도 않고, 변경될 일도 없는 변수인지라 밖으로 선언되면 좋을 것 같다는 생각이 들었어요.

컴포넌트의 복잡성을 줄여줄 수 있는 것도 있고, 매 렌더링마다 이 변수를 매번 선언하지 않아도 될 것 같아요.

ye6194 replied:
의천님 말씀대로 categoryData.js로 뺐어요!

wzrabbit commented on src/components/CategoryFilter.jsx:
```diff
@@ -0,0 +1,36 @@
+          onChange={handleChange}
+        >
+          {categoryData.map((category) => (
+            <option key={category}>{category}</option>
```
`key` prop! 👍🏻

wzrabbit commented on src/data/categoryData.js:
```diff
@@ -0,0 +1,3 @@
+const categoryData = ["한식", "중식", "일식", "양식", "아시안", "기타"];
```
`"전체"` 가 없네요? 이 변수, 그대로 export 된다음 카테고리 컴포넌트를 만드는 데 사용되지 않을까요?
서버를 켰을 때의 초기 화면인데 state 값은 "전체" 인데 카테고리 컴포넌트에는 "전체" 옵션이 없어서인지 "한식" 이 뜨는 것 같습니다 😢 해결해 주셨으면 좋겠어요 🚀 

<img src="https://github.com/user-attachments/assets/2e162b37-4c1d-4b2f-90ce-2b4db86fdb43" width="300px"/>

ye6194 replied:
이번에 급했는지... 실수가 많네요🤣 꼭 고치겠습니다!

wzrabbit reviewed (changes requested):
@ye6194 👋🏻 2️⃣ 

1, 2단계 미션 한꺼번에 구현하시느라 수고 많으셨습니다. props와 state가 이번 미션에서 추가되었네요? 이 단계부터 슬슬 재사용성을 챙길 각을 잡을 수 있는지라, 구현도 구현이지만 고민도 많이 되실 것 같아요.

정해진 정답은 없으니, UI들을 컴포넌트로 쪼개 볼지, 쪼개본다면 어떤 점을 중점으로 쪼개볼 지 등등 여러 고민을 해 보시되, 스스로 이러한 결정을 내리는 이유를 생각해보셨으면 좋겠어요.

구현할 UI 전체를 쫙 펼쳐놓고, 공통되는 부분을 찾아보는 것도 좋을 것 같아요. 공통되는 부분이 많다는 것은, 일일이 구현하지 않고 재사용하기 좋은 UI일 수 있다는 것을 의미하거든요.

1단계에서 이야기를 많이 했으니, 2단계는 올려주신 질문 두 개만 답변드리는 것으로 리뷰를 마칠게요. **1단계에도 리뷰를 달았어요!**
- [편리하게 돌아갈 수 있는 1단계 PR 버튼](https://github.com/cho-log/self-paced-react/pull/18#pullrequestreview-2362316402)

<hr>

> 저는 useEffect를 썼는데 왠지 useEffect를 쓰지 않아도 될 것 같긴 합니다. 좋은 방법이 있을까요?? 리뷰어님은 어떻게 하셨나요?

레스토랑 카테고리가 변경되면, 이를 기반으로 필터링된 레스토랑을 반환해야 하는 로직이군요?
저라면, `useEffect` 자체를 사용하지 않고, 레스토랑 목록 자체도 state로 두지 않을 것 같아요. 왜냐하면, **카테고리가 무엇인지를 알면 레스토랑의 목록이 특정되는 상황**이라, 카테고리는 계속해서 state로 두고, 레스토랑의 목록은 카테고리의 state에 따라 필터링 로직 진행 후 **계산된 값**을 그대로 반환하도록 구현해 볼 것 같아요.

React 공식 문서에서도 이와 매우 비슷한 상황을 설명하고 있는데요, useEffect를 통해 특정 state가 바뀔 경우 추가적인 state를 변경해주는 과정은 기본적으로 렌더링 사이클을 다시 한 번 발생시키고, 다시 컴포넌트에 대한 계산 작업을 해야 하는 형태가 돼요.

하지만, 레스토랑의 목록을 state를 기반으로 어떤 레스토랑이 필터링 결과로 나와야 하는지에 대한 작업은, 충분히 **렌더링 중에 수행이 가능한 연산** 이 될 수 있을 거에요. 그런 이유로 저는 이 상황에서는 `useEffect` 를 사용하지 않는 방법을 추천드리고 싶어요.

`useEffect`는, 주로 외부의 상황 변화에 따라 컴포넌트를 동기화해야 할 때 좋은 사용처가 될 수 있을 거에요.
- https://react.dev/learn/you-might-not-need-an-effect 의 첫 번째, 두 번째 예제를 확인해 보시면 도움이 많이 될 수 있을 것 같아요. 사실 저도 오늘 이 문서 보고 제가 `useEffect`를 남용하는 상황이 있다는 것도 깨닫게 되었답니다 😆 

제 방법을 코드로 간략화하여 표현하면, 이런 형태가 될 수 있을 것 같아요.

```js
const App = () => {
  const [category, setCategory] = useState('전체');
  const filteredRestaurants = filterRestaurants(category);

  const filterRestaurants = (category) => {
    // 필터링 로직
  };

  // 이후 category, filteredRestaurants를 JSX에 사용해 주세요
};

export default App;
```

> CategoryFilter.jsx에서 `console.log(category);` 이 부분이 4번이나 실행되는 이유가 뭘까요??

1. 예은님께서는 `useEffect`를 사용하셨으므로, 카테고리가 변경되면 먼저 리렌더링이 1회 일어나고, 그 다음 카테고리가 변경되었으므로 `useEffect`에 의해 필터링된 레스토랑의 정보가 변경되며 다시 리렌더링이 1회 일어납니다. 여기까지만 보면 **2회**입니다.
2. 여기서, 각 렌더링마다 렌더링은 1회가 아닌 **2회**가 일어납니다. 왜냐하면, 개발 모드일 때는 [StrictMode](https://react.dev/reference/react/StrictMode#reference) 가 활성화되어 있거든요. `index.js` 를 보시면, `<React.StrictMode>` 로 전체 컴포넌트가 감싸져있는 것을 확인하실 수 있으실 겁니다.

따라서, 카테고리가 변경되면 실행되는 리렌더링 횟수는 $2 \times 2 = 4$ 회입니다. 만약 이 프로젝트를 빌드해 실제로 배포한 상황에서는 `StrictMode`는 비활성화 되므로 $2$ 회가 돼요.

 - `StrictMode`는 개발 환경 중 잠재적인 버그들을 더 많이 잡아주기 위해, 기본적으로 프로젝트를 생성시켜주시면 포함된 채로 프로젝트가 구동되게 돼요.
 - `StrictMode`의 영향 중 하나는 컴포넌트가 **렌더링 시 2회 렌더링 된다는 점**이에요. 이는 혹시나 컴포넌트가 순수하지 못해(= 같은 입력에 대해 외부의 영향으로 다른 결과를 내놓을 수 있음) 버그가 발생하거나 다른 결과가 나오는지를 확인할 수 있도록 하는 동작이에요.
 - `StrictMode`는 개발 모드에서만 작동해요. 즉 빌드해서 실제로 프로젝트를 사용자들에게 배포한 상태에서는 비활성화 되므로 2회 렌더링도 일어나지 않아요.
 - `StrictMode`를 제거하지 않을 것을 강력히 권장해요.

답변은 여기까지 하겠습니다! 이외의 권장사항에 대해, 코멘트를 남겨드렸어요. 이에 대해 예은님의 의견을 많이 들어볼 수 있으면 좋겠네요 😊

ye6194 committed:
refactor: 코드리뷰 반영

- 불필요한 코드 삭제
- 파일 분리 (categoryInEnglish)
- export, import 방식 변경

ye6194 committed:
chore: 불필요한 설정 제거

- uninstall react-scripts

ye6194 committed:
refactor: useEffect 삭제

ye6194 commented:
@wzrabbit 

안녕하세요!

역시 공식문서가 최고네요.저도 useEffect를 남용하지 않도록 주의해야겠어요.
그래서 이번에 useEffect를 없애려고 했는데 왜... 작동이 안될까요?? `console.log()`를 해봐도 왜인지 콘솔에 아무것도 안뜨네요...
![image](https://github.com/user-attachments/assets/f0f67fd7-5957-4fd1-b9b4-ebfc3ec69560)

<br/>
<br/>

`console.log(category)`가 4번이나 나오는 이유는 되게 신기하네요!! StrictMode가 뭔지 몰랐는데 이번에 알게 되었어요.
자세한 답변 감사합니다.

<br/>
<br/>
<br/>
<br/>

> * [편리하게 돌아갈 수 있는 1단계 PR 버튼](https://github.com/cho-log/self-paced-react/pull/18#pullrequestreview-2362316402)

중간에 1단계 pr에 갈 일이 있었는데 정말 편리하네요🤭

wzrabbit reviewed (changes requested):
@ye6194 👋🏻 

구현해 주신 반영사항들을 확인했고, 반영 중 새로 생긴 문제에 대해서도 디버깅해 보았어요. 미션에서 처음에 개발환경 설정이 되어 있는 Vite 개발환경으로 되돌리시느라 수고 많으셨습니다.

새로 생긴 문제의 경우에는 해결을 위해 아래의 두 힌트를 드리고자 해요. 힌트를 기반으로 문제를 해결해 보셨으면 좋겠습니다.

1. CRA의 `react-scripts`로 실행하던 개발 서버를 Vite를 이용해 실행하는 것으로 변경하셨음에 주목해 보는 것은 어떨까요? 명령어도 `npm run start`(= `react-scripts start`) 에서 `npm run dev`(= `vite`)로 변경하셨을 거에요.
2. [1단계 미션](https://github.com/cho-log/self-paced-react/pull/18) 을 진행하실 때 여러 에러들을 마주하셨을 거에요. 어떤 에러가 발생했었고, 왜 에러가 발생했었을까요?

ye6194 committed:
fix: CRA에서 Vite로 제대로 바꿔지지 않았던 문제 해결

- 화면은 그려지지만 개발자 도구의 콘솔에 아무것도 뜨지 않고 필터가 작동하지 않는 문제가 있었음
- index.js 파일 삭제
- /public/index.html을 루트로 옮김
- 원래 루트에 있던 index.html을 templates 폴더로 옮김
- public에 있던 .css, .png 파일들을 templates 폴더로 옮김
- App.jsx의 import문에서 확장자 명시

ye6194 commented:
@wzrabbit 
문제는 해결한 것 같아요!
CRA에서 Vite로 제대로 마이그레이션이 되지 않은게 원인인 것 같긴 한데 해결하려고 이것저것 많이 바꿔서 뭐가 원인이었는지는 확실하게 모르겠네요ㅠㅠ...
문제를 해결하기 위해서 한 일은 커밋 내역에 적어두었어요!

wzrabbit commented on src/App.jsx:
```diff
@@ -1,7 +1,35 @@
 import "./App.css";
+import { useState } from "react";
+import Header from "./components/Header.jsx";
+import CategoryFilter from "./components/CategoryFilter.jsx";
+import RestaurantList from "./components/RestaurantList.jsx";
+import AddRestaurantModal from "./components/AddRestaurantModal.jsx";
+import RestaurantDetailModal from "./components/RestaurantDetailModal.jsx";
+import restaurantsData from "./data/restaurantsData.js";
```
```suggestion
import Header from "./components/Header";
import CategoryFilter from "./components/CategoryFilter";
import RestaurantList from "./components/RestaurantList";
import AddRestaurantModal from "./components/AddRestaurantModal";
import RestaurantDetailModal from "./components/RestaurantDetailModal";
import restaurantsData from "./data/restaurantsData";
```
확장자를 명시하지 않으셔도 괜찮답니다😊 아래의 링크를 참고해 보세요!

- https://vite.dev/config/shared-options#resolve-extensions

ye6194 committed:
refactor: 코드 리뷰 반영

- categoryData, categoryInEnglish를 UPPER_SNAKE_CASE로 변경
- CATEGORY_IN_ENGLISH을 RestaurantList.jsx로 이동

wzrabbit reviewed (approved):
@ye6194 👋🏻 
트러블슈팅 하시느라 고생 많으셨습니다. 문제를 해결하셨다니 다행이네요. 이후 자잘하게 코드리뷰 반영하신 것도 잘 확인했습니다. 커밋 메시지도 자세하네요.

문제를 해결하셨지만 원인에 대해서는 긴가민가하신 것 같아서, 이번에 짚고 넘어갈게요.

> 화면은 그려지지만 개발자 도구의 콘솔에 아무것도 뜨지 않고 필터가 작동하지 않는 문제가 있었음

라고 현재 발생하시는 문제를 설명해 주셨는데, 이 이유는 작성하신 코드의 로직이 잘못되어서가 아니라 단순히 서버에서 실행된 **HTML 파일이 미션 구현 전의 탬플릿 파일이어서** 그런 것입니다. 그러니 아무 에러도 없지만 아무 기능도 작동하지 않는 껍데기 HTML 파일처럼 작동한 것이죠.

힌트에서 제가 1단계 미션의 피드백과 함께 어떤 에러가 발생했는지를 확인해 보라고 요청드렸었어요. 이번 문제에서는 아래의 요인이 관련이 있었답니다.
- `public` 안에 `index.html` 이 없다는 에러
- `index.js` 가 `src` 안에 없다는 에러

요약하면 **초기에 세팅된 Vite 프로젝트에서 사용하는 파일들과 CRA 프로젝트에서 사용하는 파일들의 위치가 달라서** 이 문제가 생긴 것이었습니다. 따라서 문제를 해결하기 위해 파일들의 경로를 올바르게 바꾸는 변경사항을 내셨다면 올바르게 문제를 해결하신 것입니다.

이렇게 하면 `App` 파일에서 자잘한 에러가 발생하나 에러 메시지가 명확하므로 `console.log` 만 제거하면 올바르게 작동할 것이고, 실제로도 그렇게 하신 것을 확인했습니다. 💯 

앞으로도 문제 해결을 위해 여러 시도들을 해 보시기를 바래요! 긴 기간 동안 미션 수행하시느라 수고 많이 하셨습니다.

ye6194 committed:
push

ye6194 committed:
Merge remote-tracking branch 'upstream/ye6194' into step2
